compiler-core/src/build.rs

```diff
@@ -183,8 +184,9 @@ impl Module {
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub enum Located<'a> {
```

Comment:
Having this use references creates issues with stuff that are in RefCells.
So far, it created issues with the `Type` of `Pattern(TypedPattern, Type)`.
Is it a big deal having copies here?

Reply:
I would prefer not to copy. Where is it that we have the Rc problem? I couldn't see it at a glance

Reply:
https://github.com/gleam-lang/gleam/blob/cd3798f68aee1fb094c9768593e42519548fc460/compiler-core/src/ast.rs#L1079-L1091


Specifically, the link inside `TypeVar::Link { type_ }`